### ISSUE TITLE
Copy stops on first non-updated file when using :update true

### ISSUE THREAD

**Issue by robert914** [m0smith/lein-resource#24]
When setting the :update true flag, the copy appears to stop after the first file it hits that doesn't need an update. So for example:

1) I have 5 files that lein-resource copies
2) I run "lein resource" once, all 5 files are copied
3) I run "lein resource" again, no files are copied.
4) I edit the source of the first and fourth files.
5) I run "lein resource" again, only the first file is copied, the fourth file is not.
6) I run "lein resource" again, no files are copied.

You can see this if you enable :verbose true, and grep the output for "copy-file-spec", you'll see that it stops doing the copies after the first non-updated file. I would expect that each file pair should be analyzed, and not exit on the first one that doesn't need updating.

Basically, I believe what is happening is the laziness of this part of the code:

`  (every? identity (->> (file-spec-seq project-info)
                        (verbose-msg project-info "file-spec-seq")
                        (map (partial copy-file-spec project-info))))`

Causes it to exit the "every?" loop the first time something doesn't need updating, because it only returns a vector pair of files when the file is copied, but not when it doesn't need updating. Thus, the "every?" stops at the first "failure" (i.e. nil value) instead of proceeding.

Possibly the fix is that you always return the pair of files, regardless of if they were "updated" or not.

**Comment by m0smith** [comment]
Thanks for bringing this to my attention.  I will try and get it fixed over the next few days,

**Comment by robert914** [comment]
Thanks, I'd really appreciate it!

BTW, I was thinking, I wonder if the fix is as simple as just wrapping that "every?" form in a "doall" form? I haven't tested that theory yet, but it might solve it without breaking anything downstream of the copy-task function.

**Comment by robert914** [comment]
I tried a few things, and this appears to solve the issue:

`(every? identity (doall (->> (file-spec-seq project-info)
                        (verbose-msg project-info "file-spec-seq")
                        (map (partial copy-file-spec project-info)))))`

So adding the "doall" inside fixes the problem as far as my testing goes, but not sure that's the solution you want. Just FYI.

**Comment by m0smith** [comment]
I am having the same problem with clean which has similar code.  However I cannot figure out why the test is failing.  I will see what I can do over the weekend.

If you want to make a pull request with your change I will grab it as well.

**Comment by robert914** [comment]
I just copied your "resource.clj" to my project locally and made the above change (added "doall"). I think the clean has the same "every?" structure, you can likely fix it with the same thing, however, I have not tested / played with using the clean that way myself.

**Comment by m0smith** [comment]
I pushed version 17.06.1 to clojars.  Take a look.

**Comment by robert914** [comment]
Great! Seems to work fine for me. Thanks for the update!
